packages/mermaid/src/diagrams/flowchart/flowDb.ts

```diff
@@ -16,1101 +18,1082 @@ import {
-        rawEdge?.stroke === 'invisible' || rawEdge?.type === 'arrow_open' ? 'none' : arrowTypeStart,
-      arrowTypeEnd:
-        rawEdge?.stroke === 'invisible' || rawEdge?.type === 'arrow_open' ? 'none' : arrowTypeEnd,
-      arrowheadStyle: 'fill: #333',
-      cssCompiledStyles: getCompiledStyles(rawEdge.classes),
-      labelStyle: styles,
-      style: styles,
-      pattern: rawEdge.stroke,
-      look: config.look,
-      animate: rawEdge.animate,
-      animation: rawEdge.animation,
-    };
+    const e = this.getEdges();
+    e.forEach((rawEdge, index) => {
+      const { arrowTypeStart, arrowTypeEnd } = this.destructEdgeType(rawEdge.type);
+      const styles = [...(e.defaultStyle ?? [])];
 
-    edges.push(edge);
-  });
+      if (rawEdge.style) {
+        styles.push(...rawEdge.style);
+      }
+      const edge: Edge = {
+        id: getEdgeId(rawEdge.start, rawEdge.end, { counter: index, prefix: 'L' }, rawEdge.id),
+        start: rawEdge.start,
+        end: rawEdge.end,
+        type: rawEdge.type ?? 'normal',
+        label: rawEdge.text,
+        labelpos: 'c',
+        thickness: rawEdge.stroke,
+        minlen: rawEdge.length,
+        classes:
+          rawEdge?.stroke === 'invisible'
+            ? ''
+            : 'edge-thickness-normal edge-pattern-solid flowchart-link',
+        arrowTypeStart:
+          rawEdge?.stroke === 'invisible' || rawEdge?.type === 'arrow_open'
+            ? 'none'
+            : arrowTypeStart,
+        arrowTypeEnd:
+          rawEdge?.stroke === 'invisible' || rawEdge?.type === 'arrow_open' ? 'none' : arrowTypeEnd,
+        arrowheadStyle: 'fill: #333',
+        cssCompiledStyles: this.getCompiledStyles(rawEdge.classes),
+        labelStyle: styles,
+        style: styles,
+        pattern: rawEdge.stroke,
+        look: config.look,
+        animate: rawEdge.animate,
+        animation: rawEdge.animation,
+      };
+
+      edges.push(edge);
+    });
 
-  return { nodes, edges, other: {}, config };
-};
+    return { nodes, edges, other: {}, config };
+  };
 
-export default {
-  defaultConfig: () => defaultConfig.flowchart,
-  setAccTitle,
-  getAccTitle,
-  getAccDescription,
-  getData,
-  setAccDescription,
-  addVertex,
-  lookUpDomId,
-  addLink,
-  updateLinkInterpolate,
-  updateLink,
-  addClass,
-  setDirection,
-  setClass,
-  setTooltip,
-  getTooltip,
-  setClickEvent,
-  setLink,
-  bindFunctions,
-  getDirection,
-  getVertices,
-  getEdges,
-  getClasses,
-  clear,
-  setGen,
-  defaultStyle,
-  addSubGraph,
-  getDepthFirstPos,
-  indexNodes,
-  getSubGraphs,
-  destructLink,
-  lex,
-  exists,
-  makeUniq,
-  setDiagramTitle,
-  getDiagramTitle,
-};
+  public defaultConfig = () => defaultConfig.flowchart;
+  public setAccTitle = setAccTitle;
+  public setAccDescription = setAccDescription;
+  public setDiagramTitle = setDiagramTitle;
+  public getAccTitle = getAccTitle;
+  public getAccDescription = getAccDescription;
+  public getDiagramTitle = getDiagramTitle;
```

Comment:
Hhhhmmmmmmmmmmmm, this means part of the DB will be tied to a diagram, and some parts will change based on the last rendered diagram.